dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -813,17 +812,27 @@ private WorkflowVersion setupWorkflowFilesForGitHubVersion(Triple<String, Date,
                 }
 
                 if (missingParamFiles.size() > 0) {
-                    validationMessage.put(DOCKSTORE_YML_PATH, String.format("%s: %s.", missingParamFiles.size() == 1 ? "The following file is missing" : "The following files are missing",
-                            missingParamFiles.stream().map(paramFile -> String.format("'%s'", paramFile)).collect(Collectors.joining(", "))));
+                    validationMessage = String.format("The following %s missing: %s.", missingParamFiles.size() == 1 ? "file is" : "files are",
+                            missingParamFiles.stream().map(paramFile -> String.format("'%s'", paramFile)).collect(Collectors.joining(", ")));
                 }
             }
         } else {
             // File not found or null
             LOG.info("Could not find the file " + primaryDescriptorPath + " in repo " + repository);
-            validationMessage.put(DOCKSTORE_YML_PATH, "Could not find the primary descriptor file '" + primaryDescriptorPath + "'.");
+            validationMessage = "Could not find the primary descriptor file '" + primaryDescriptorPath + "'.";
+        }
+
+        try {
+            DockstoreYamlHelper.validateDockstoreYamlProperties(dockstoreYml.getContent()); // Validate that there are no unknown properties
+        } catch (DockstoreYamlHelper.DockstoreYamlException ex) {
+            validationMessage = validationMessage.isEmpty() ? ex.getMessage() : validationMessage + " " + ex.getMessage();
```

Comment:
The newline doesn't show up in the UI, so it would require a UI PR to have it show up properly. If it did show up properly, it would look like this:
![image](https://user-images.githubusercontent.com/25287123/142065250-4419bae8-00ff-41d5-96a0-d27d7db26837.png)

Should I make this change?

Reply:
Not sure about best practice here. Doesn't seem like the webservice should be doing this kind of formatting 
@coverbeck @garyluu thoughts?

Reply:
Agreed, it feels wrong for the web service to do formatting like that. Also, I don't think a newline would even do anything on its own to the UI, you'd need to instead make it a `<br>` (which feels even worse), or have the UI convert newlines to `<br>`.

Can you create two validation objects? I don't know the code that well -- does the web service allow more than one validation for a single file? Will the UI handle it?

Reply:
As is, you can't create two validation objects for a file type, which was the reason why I had to append the unknown property validation message to any existing ones. The validation message is stored as a `Map<String, String>` which maps the file path to the validation message. I could try changing it to `Map<String, List<String>>` so we can store multiple validation messages for a file.

Reply:
Created ticket: https://github.com/dockstore/dockstore/issues/4582